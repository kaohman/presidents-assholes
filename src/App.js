import React, { Component } from 'react';
import './App.css';


class App extends Component {

  async componentDidMount() {
    
  }

  render() {
    return (
      <div>
      </div>
    );
  }
}

export default App;
