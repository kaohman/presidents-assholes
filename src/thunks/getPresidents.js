import { setPresidents, setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getPresidents = () => {
  return async (dispatch) => {
    dispatch(setLoading(true));
    try {
      const presidents = await fetchData('http://localhost:3001/api/v1/presidents');
      dispatch(setPresidents(presidents));
    } catch (error) {
      dispatch(setError(error));
    }
    dispatch(setLoading(false));
  }
}
