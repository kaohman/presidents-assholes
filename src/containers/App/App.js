import React, { Component } from 'react';
import { getPresidents } from '../../thunks/getPresidents';
import { connect } from 'react-redux';
import Presidents from '../Presidents/Presidents';
import FilterControls from '../FilterControls/FilterControls';


export class App extends Component {

  componentDidMount() {
    this.props.getPresidents();
  }

  render() {
    const { loading, error } = this.props;
    return (
      <div>
        <header>
          <h1>Presidents Assholes</h1>
          {loading && <h2>Loading data...</h2>}
          {error && <h2>Error fetching data: {error}</h2>}
        </header>
        <FilterControls />
        <Presidents />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  error: state.error,
  loading: state.loading,
});

export const mapDispatchToProps = (dispatch) => ({
  getPresidents: () => dispatch(getPresidents()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
