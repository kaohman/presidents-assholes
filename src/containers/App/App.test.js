import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getPresidents } from '../../thunks/getPresidents';
jest.mock('../../thunks/getPresidents');
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let getPresidentsMock;

  beforeEach(() => {
    getPresidentsMock = jest.fn();
    wrapper = shallow(
      <App 
        getPresidents={getPresidentsMock}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should call getPresidents', () => {
      wrapper.instance().componentDidMount();
      expect(getPresidentsMock).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with a presidents array', () => {
      const mockState = {
        error: 'error message',
        loading: false,
        otherState: false
      };
      const expected = {
        error: 'error message',
        loading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with getPresidents is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getPresidents();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getPresidents();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

    });
  });
});