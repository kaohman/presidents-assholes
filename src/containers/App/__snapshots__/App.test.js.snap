// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <header>
    <h1>
      Presidents Assholes
    </h1>
  </header>
  <Connect(FilterControls) />
  <Connect(Presidents) />
</div>
`;
